File: FromSlides/Fortran/21_set_threads_in_program.f90
PROGRAM hello_world

  USE omp_lib

  CALL omp_set_num_threads(4)

  PRINT *, 'Hello world from the main program'

  !$OMP PARALLEL
    PRINT '(A,I0)','Hello world! I am processor ', omp_get_thread_num()
  !$OMP END PARALLEL

END PROGRAM hello_world

File: FromSlides/C/21_set_threads_in_program.c
#include <stdio.h>
#include <omp.h>

int main()
{
  printf("Hello world from the main program\n");
  omp_set_num_threads(4);
#pragma omp parallel
  {
    printf("Hello world! I am processor %i\n", omp_get_thread_num());
  }
  return 0;
}

